truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/TruffleLanguage.java

```diff
@@ -386,6 +386,19 @@ public Object importSymbol(String globalName) {
         }
 
         /**
+         * Produces an array of strings for the MIME types supported by this {@link PolyglotEngine}.
+         * There may be multiple MIME types for a single language. Each MIME type will appear only
+         * once, even if multiple language declare it.
+         *
+         * MIME types from this array can be used in methods such as {@link #parse}.
+         *
+         * @return a non-null array of a string for each supported MIME type
+         */
+        public String[] getSupportedMimeTypes() {
+            return API.getSupportedMimeTypes(vm);
```

Comment:
As far as I understand your use-case a method like
```
public boolean isMimeTypeSupported(String mimeType);
```
would work for you as well. If so, I prefer it to the array of all mime types.